modules/standard/IO.chpl

```diff
@@ -2772,11 +2796,10 @@ This function is equivalent to calling :proc:`open` and then
               corresponding parameter of the :record:`channel` type.
               Defaults to true, but when safe, setting it to false
               can improve performance.
-:arg start: zero-based byte offset indicating where in the file the
-            channel should start reading. Defaults to 0.
-:arg end: zero-based byte offset indicating where in the file the
-          channel should no longer be allowed to read. Defaults
-          to a ``max(int(64))`` - meaning no end point.
+:arg region: zero-based byte offset indicating where in the file the
```

Comment:
Do we have any tests for the case where the lower bound is less than zero? Do the error message in that case still make sense (i.e., it doesn't refer to a 'start' argument?)

Reply:
I'm not aware of any, but I can definitely add some

Reply:
Interesting.  It looks like it doesn't fail on the `openreader` call but does fail when we try to do something with it:
```
uncaught SystemError: Invalid argument (in channel.readLine(ref s: string) with path "openreaderLimited.txt" offset -3)
  openreaderLessThan0.chpl:6: thrown here
  openreaderLessThan0.chpl:6: uncaught here
```
Seems like some checks are in order, but probably ones that get turned off with `--fast` and `--no-bounds-checks` or whatever the flag is.  Do you think I should check with the rest of the team on those or do they seem sufficiently obvious to just go ahead and add?

Reply:
If we know that we're going to always reject a lower bounds less than zero, then I'd prefer we throw an error in the reader-creator functions/methods. I don't think we should turn them off for ``--fast`` or ``--no-bounds-checks`` because they don't require any costly checks (literally ``< 0``).

I think the question of if/when we should throw for an upper out of bounds error is worthy of an issue. For example, if the user specifies a region that exceeds the file length (which is perhaps discovered or only finalized well after the reader is created), is that an error?